File: innn/intersect-android/app/src/main/java/com/tmmt/innersect/ui/fragment/BaseFragment.java
package com.tmmt.innersect.ui.fragment;


import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;

import butterknife.ButterKnife;

/**
 * Created by Administrator on 2016/10/8.
 */
public abstract class BaseFragment extends Fragment  {

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setHasOptionsMenu(true);
    }
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view= inflater.inflate(getLayout(),container,false);
        ButterKnife.bind(this,view);
        initView(view);
        return view;
    }
    protected void initView(View view){

    }


    abstract int getLayout();
    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if(item.getItemId()==android.R.id.home){
            getActivity().onBackPressed();
        }
        return true;
    }
    @Override
    public void onDestroy() {
        super.onDestroy();
    }
}

File: app/src/main/java/com/stesso/android/BaseFragment.kt
package com.stesso.android

import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.app.AlertDialog
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.stesso.android.datasource.net.ApiService
import com.stesso.android.di.component.FragmentComponent
import com.stesso.android.di.module.FragmentModule
import com.stesso.android.model.Account
import com.stesso.android.model.RootNode
import com.stesso.android.utils.applySingleSchedulers
import com.stesso.android.utils.createProgressDialog
import com.stesso.android.utils.toast
import io.reactivex.Single
import io.reactivex.disposables.CompositeDisposable
import javax.inject.Inject

open class BaseFragment : Fragment() {

    private val disposableContainer = CompositeDisposable()
    @Inject
    lateinit var apiService: ApiService

    lateinit var progressDialog: AlertDialog

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        progressDialog = context.createProgressDialog()

    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        return inflater.inflate(getLayoutId(), container, false)


    }

    protected open fun getLayoutId() = R.layout.fragment_mine


    protected fun getFragmentComponent(): FragmentComponent {
        return App.instance().component.plus(FragmentModule())
    }

    protected fun <T> doHttpRequest(single: Single<RootNode<T>>, quite: Boolean = false, onSuccess: (T?) -> Unit) {
        progressDialog.show()
        val disposable = single.compose(applySingleSchedulers())
                .doFinally {
                    progressDialog.dismiss()
                }
                .subscribe({ rootNode ->
                    if (rootNode.errno != 0) {
                        if (rootNode.errno == 501) {
                            Account.logout()
                        }
                        if (!quite) {
                            context?.toast(rootNode.errmsg ?: "")
                        }
                    } else {
                        onSuccess(rootNode.data)
                    }
                }, {
                    it.printStackTrace()
                })
        disposableContainer.add(disposable)
    }

    override fun onDestroy() {
        disposableContainer.dispose()
        super.onDestroy()
    }

}
